src/main/java/org/aksw/simba/lemming/ColouredGraphDecorator.java

```diff
@@ -0,0 +1,300 @@
+/**
+ * 
+ */
+package org.aksw.simba.lemming;
+
+import org.aksw.simba.lemming.mimicgraph.constraints.TripleBaseSingleID;
+import org.aksw.simba.lemming.util.IntSetUtil;
+
+import com.carrotsearch.hppc.BitSet;
+
+import it.unimi.dsi.fastutil.ints.IntArrayList;
+import it.unimi.dsi.fastutil.ints.IntSet;
+
+/**
+ * Base Decorator for Coloured Graph Class
+ * 
+ * @author Pranav
+ */
+public class ColouredGraphDecorator implements IColouredGraph {
+
+    protected IColouredGraph graph;
+    protected boolean isAddingEdge;
+    protected TripleBaseSingleID triple;
```

Comment:
Are these two attributes really a part of a generic decorator? Wouldn't it make more sense to have a generic decorator class that handles the decorator functionality and an additional class that handles the specific case in which somebody wants to change a single edge? :thinking:

Reply:
The two attributes are not used in this base decorator. However, they are needed in the AddEdgeDecorator and RemoveEdgeDecorator classes which inherit these attributes and their get(), set() methods. 

Removing them from here would mean they would be added to both these sub classes. I just included them here to not to repeat the code in both sub classes.

Reply:
Fair point. Typically, I would argue to add another class in-between that offers this functionality. However, I think we can go with your solution as well :+1: